diff --git a/mt7996/pci.c b/mt7996/pci.c
index b7d9193..5315bad 100644
--- a/mt7996/pci.c
+++ b/mt7996/pci.c
@@ -241,6 +241,11 @@ static void mt7996_pci_remove(struct pci_dev *pdev)
 	mt7996_unregister_device(dev);
 }
 
+static void mt7996_pci_shutdown(struct pci_dev *pdev)
+{
+	mt7996_pci_remove(pdev);
+}
+
 struct pci_driver mt7996_hif_driver = {
 	.name		= KBUILD_MODNAME "_hif",
 	.id_table	= mt7996_hif_device_table,
@@ -253,6 +258,7 @@ struct pci_driver mt7996_pci_driver = {
 	.id_table	= mt7996_pci_device_table,
 	.probe		= mt7996_pci_probe,
 	.remove		= mt7996_pci_remove,
+	.shutdown	= mt7996_pci_shutdown,
 };
 
 MODULE_DEVICE_TABLE(pci, mt7996_pci_device_table);
